src/compiler/scala/tools/nsc/transform/Fields.scala

```diff
@@ -360,9 +364,12 @@ abstract class Fields extends InfoTransform with ast.TreeDSL with TypingTransfor
           ClassInfoType(parents, allDecls, clazz)
         } else tp
 
+
+      case tp@ClassInfoType(parents, oldDecls, clazz) if !needsMixin(clazz) => tp
```

Comment:
I'm not sure this will fly, since we also need to hit the next case if there's a module or lazy val that needs expansion, for example.